xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/KmsEventDecryptor.java

```diff
@@ -79,14 +79,18 @@ public String decryptAsymmetric(String encoded) {
       return encoded;
     }
 
-    ByteBuffer cipherText = ByteBuffer.wrap(Base64.getDecoder().decode(encoded.getBytes()));
-    DecryptRequest req = new DecryptRequest()
-        .withKeyId(kmsKeyArn)
-        .withEncryptionAlgorithm(ASYMMETRIC_ALGORITHM)
-        .withCiphertextBlob(cipherText);
+    String tmp = encoded.substring(2, encoded.length() - 2);
```

Comment:
encode.length() > 1 is asserted here

Reply:
We check now if the string is encrypted. This means it is not null, contains markers at beginning and end, so it is at least 4 chars long.